File: Languages/Csharp/BeeMovie.cs
using System;
using System.IO;
using System.Collections.Generic;
using System.Linq;

public class BeeMovie
{
    public static void Main(string[] args)
    {
        // Hello, World!
        Console.WriteLine("Hello, World!");
        AnalyzeBeeMovie();
    }
    
    private static void AnalyzeBeeMovie()
    {
        string scriptPath = "../beemoviescript.txt";
        
        if (!File.Exists(scriptPath))
        {
            Console.WriteLine($"File not found: {scriptPath}");
            return;
        }
        
        Console.WriteLine("Bee Movie Script:");
        Console.WriteLine("--------------------------------------------------");
        
        int[] letterCounts = new int[26];
        int totalLetters = 0;
        
        try
        {
            string[] lines = File.ReadAllLines(scriptPath);
            
            // Print each line
            foreach (string line in lines)
            {
                Console.WriteLine(line);
                
                // Count letters
                foreach (char c in line)
                {
                    if (char.IsLetter(c))
                    {
                        char lower = char.ToLower(c);
                        letterCounts[lower - 'a']++;
                        totalLetters++;
                    }
                }
            }
        }
        catch (Exception e)
        {
            Console.WriteLine($"Error reading file: {e.Message}");
            return;
        }
        
        Console.WriteLine("--------------------------------------------------");
        Console.WriteLine("Analysis complete.");
        
        if (totalLetters > 0)
        {
            // Create list of letter counts
            var counts = new List<LetterCount>();
            for (int i = 0; i < 26; i++)
            {
                counts.Add(new LetterCount((char)('a' + i), letterCounts[i]));
            }
            
            // Sort by count (descending)
            counts = counts.OrderByDescending(x => x.Count).ToList();
            
            Console.WriteLine("\nTop 3 most commonly used letters:");
            for (int i = 0; i < 3 && counts[i].Count > 0; i++)
            {
                Console.WriteLine($"{i + 1}. '{counts[i].Letter}': {counts[i].Count} times");
            }
        }
        else
        {
            Console.WriteLine("No letters found in the script.");
        }
    }
    
    class LetterCount
    {
        public char Letter { get; }
        public int Count { get; }
        
        public LetterCount(char letter, int count)
        {
            Letter = letter;
            Count = count;
        }
    }
}

File: Languages/PHP/bee_movie.php
<?php
// Hello, World!
echo "Hello, World!\n";

function analyzeBeeMovie() {
    $scriptPath = "../beemoviescript.txt";
    
    if (!file_exists($scriptPath)) {
        echo "File not found: $scriptPath\n";
        return;
    }
    
    echo "Bee Movie Script:\n";
    echo "--------------------------------------------------\n";
    
    $letterCounts = array_fill_keys(range('a', 'z'), 0);
    $totalLetters = 0;
    
    // Read file line by line and print each line
    $lines = file($scriptPath, FILE_IGNORE_NEW_LINES);
    foreach ($lines as $line) {
        echo $line . "\n";
        
        // Count letters
        for ($i = 0; $i < strlen($line); $i++) {
            $char = $line[$i];
            if (ctype_alpha($char)) {
                $lower = strtolower($char);
                $letterCounts[$lower]++;
                $totalLetters++;
            }
        }
    }
    
    echo "--------------------------------------------------\n";
    echo "Analysis complete.\n";
    
    if ($totalLetters > 0) {
        // Sort by count (descending)
        arsort($letterCounts);
        
        echo "\nTop 3 most commonly used letters:\n";
        $count = 0;
        foreach ($letterCounts as $letter => $count) {
            if ($count > 0 && $count < 3) {
                echo ($count + 1) . ". '$letter': $count times\n";
                $count++;
            }
            if ($count >= 3) break;
        }
    } else {
        echo "No letters found in the script.\n";
    }
}

analyzeBeeMovie();
?>
